Bluepill-cli/Bluepill-cli/Bluepill/Bluepill.m

```diff
@@ -240,8 +239,10 @@ - (void)createSimulatorWithContext:(BPExecutionContext *)context {
         // If we failed to create the simulator, there's no reason for us to try to delete it, which can just cause more issues
         if (--__self.maxCreateTries > 0) {
```

Comment:
>Are you seeing a case where you're hitting create and the simulator already exists? What exactly is the flow that leads to that?

The method isn't very accurately named, maybe I should change that as part of this PR. `createSimulator` actually creates the simulator, launches the app, and waits until the simulator gets into a ready state. In our logs where this was failing, I would see this log printed out (https://github.com/linkedin/bluepill/blob/master/Bluepill-cli/Bluepill-cli/Simulator/BPSimulator.m#L86), suggesting that the creation of the simulator was actually successful.

It might make sense to try and retry the app launch or something in this case instead, but the goal of this PR is to stop leaking resources. :)